### ISSUE TITLE
Support for polars dataframes

### ISSUE THREAD

**Issue by muendlein** [holoviz/datashader#1199]
As [Polars](https://github.com/pola-rs/polars) is becoming more and more popular I am wondering if there are any consideration to support its dataframes. Given the performance oriented focus of this package, it might be solid contender.

**Comment by ianthomas23** [comment]
Yes, `polars` is certainly something we would like to add support for.

**Comment by ivirshup** [comment]
Other plotting libraries have been adding support for polars (and others) by supporting the `__dataframe__` interchange protocol (https://data-apis.org/dataframe-protocol/latest/index.html)

* https://github.com/altair-viz/altair/pull/2888
* https://github.com/mwaskom/seaborn/pull/3369

I think this would be a great solution here as it would allow polars as well as other tabular libraries to be used with data shader. 

My use case would be to provide a __dataframe__ interface to plotting libraries from an [`AnnData`](https://github.com/scverse/anndata) object. This use case would rely on data shader only requesting the columns it actually needs for a plot, and not try to request every column of the dataframe..

**Comment by ianthomas23** [comment]
Yes, it is a good idea. It just needs someone to volunteer to do the required work.

**Comment by hathawayj** [comment]
I wonder if [Narwhals](https://github.com/narwhals-dev/narwhals) could be leveraged?
